plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/OpenShiftConnector.java

```diff
@@ -311,15 +318,21 @@ private void addContainer(IDeploymentConfig dc, String workspaceID, String image
         dc.getContainer(containerName).setImagePullPolicy(IMAGE_PULL_POLICY_IFNOTPRESENT);
 
         ModelNode dcFirstContainer = ((DeploymentConfig)dc).getNode().get("spec").get("template").get("spec").get("containers").get(0);
+        
+        boolean isDevMachine = isDevMachine(exposedPorts);
+        // Run non-dev machine as root user
+        String UID = isDevMachine ? UID_USER : UID_ROOT;
 
-        // Run as root user
         dcFirstContainer.get("securityContext").get("privileged").set(true);
-        dcFirstContainer.get("securityContext").get("runAsUser").set("1000");
+        dcFirstContainer.get("securityContext").get("runAsUser").set(UID);
 
         // LivenessProbe
-        dcFirstContainer.get("livenessProbe").get("tcpSocket").get("port").set(4401);
-        dcFirstContainer.get("livenessProbe").get("initialDelaySeconds").set(120);
-        dcFirstContainer.get("livenessProbe").get("timeoutSeconds").set(1);
+        // TODO: CHE-53 Improve workspaces liveness probe for non-dev machines
+        if (isDevMachine) {
+            dcFirstContainer.get("livenessProbe").get("tcpSocket").get("port").set(4401);
```

Comment:
Now that you have defined constant CHE_WORKSPACE_AGENT_PORT you could use it here too.